src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -740,13 +740,16 @@ async def get_exporting_owner(owned_dependency: OwnedDependency) -> ExportedTarg
                 sibling = next(exported_ancestor_iter, None)
             if sibling_owners:
                 raise AmbiguousOwnerError(
-                    f"Exporting owners for {target.address} are "
-                    f"ambiguous. Found {exported_ancestor.address} and "
-                    f"{len(sibling_owners)} others: "
+                    f"python_distribution target owning {target.address} is ambiguous. "
```

Comment:
We don't technically know it's a `python_distribution`. This would be misleading with subclasses. You could call `target.alias` instead.

Reply:
`target.alias` is the alias of the target whose owner we're seeking, which is typically a `python_library`. The owner is a `python_distribution` (or, technically, a subclass, true, but I would rather sacrifice a tiny bit of accuracy in the general case for clarity in the usual case).

Reply:
Can we call `exported_ancestor.alias`?

Reply:
Maybe, but what if the siblings aren't of this type? And anyway, this is a pretty rare corner case. Doesn't seem worth the worry.